src/doc/unstable-book/src/library-features/asm.md

```diff
@@ -563,13 +565,16 @@ Some registers have multiple names. These are all treated by the compiler as ide
 | RISC-V | `f[10-17]` | `fa[0-7]` |
 | RISC-V | `f[18-27]` | `fs[2-11]` |
 | RISC-V | `f[28-31]` | `ft[8-11]` |
+| Hexagon | `r29` | `sp` |
+| Hexagon | `r30` | `fr` |
+| Hexagon | `r31` | `lr` |
 
 Some registers cannot be used for input or output operands:
 
 | Architecture | Unsupported register | Reason |
 | ------------ | -------------------- | ------ |
 | All | `sp` | The stack pointer must be restored to its original value at the end of an asm code block. |
-| All | `bp` (x86), `r11` (ARM), `x29` (AArch64), `x8` (RISC-V) | The frame pointer cannot be used as an input or output. |
+| All | `bp` (x86), `r11` (ARM), `x29` (AArch64), `x8` (RISC-V), `fr` (Hexagon) | The frame pointer cannot be used as an input or output. |
```

Comment:
All registers can be used as inputs on Hexagon, that is the architecture doesn't not differentiate between `r0` and `r29` or `r30`, except for a small set of instructions which require fewer bits to encode registers.  As a matter of fact, most loads/stores to stack are done via `r29` or `r30` (i.e. `sp` or `fp`), and a return from a function that does not allocate stack space is done via `jumpr r31`.

All of these registers can also be modified, but that's typically a bad idea.